sad-comparisons.py

```diff
@@ -126,30 +137,24 @@ def model_comparisons(raw_data, dataset_name, data_dir, cutoff = 9):
             
             #Convert relative likelihoods to list
             relative_likelihoods_output = relative_likelihoods.tolist() 
-            
-            # Inserts a blank in the output if the Poisson lognormal returned -inf
-            if pln_blank == 1:
-                weights_output.insert(1, '')
-                likelihood_list.insert(1, '')
-                relative_likelihoods_output.insert(1, '')
-            
-            # Inserts a blank in the output if the negative binomial exceeded the max number of iterations
-            if negbin_blank == 1:
-                weights_output.insert(2, '')
-                likelihood_list.insert(2, '')
-                relative_likelihoods_output.insert(2, '')
-                                    
+                                                
             # Format results for output
             for weight in weights_output:
                 results1 = [[site, S, N] + weights_output]
             results2 = [[site, S, N] + likelihood_list]
             results3 = [[site, S, N] + relative_likelihoods_output]
-                                        
+            results.append([site, S, N] + weights_output  + likelihood_list + relative_likelihoods_output)
+
             # Save results to a csv file:
             output1.writerows(results1)
             output2.writerows(results2)
             output3.writerows(results3)
-    
+
+    results = DataFrame(results, columns=['site', 'S', 'N', 'AICc_logseries', 'AICc_pln', 'AICc_negbin',
+                                         'AICc_zipf', 'likelihood_logseries', 'likelihood_pln',
+                                         'likelihood_negbin', 'likelihood_zipf', 'relative_ll_logseries',
+                                         'relative_ll_pln', 'relative_ll_negbin', 'relative_ll_zipf'])
+    results.to_csv(os.path.join(data_dir, dataset_name +  '_likelihood_results.csv'), index=False)
```

Comment:
Thanks! That makes sense.

On Fri, Feb 26, 2016 at 1:58 PM, Ethan White <notifications@github.com>
wrote:

> In sad-comparisons.py
> <https://github.com/weecology/sad-comparison/pull/189#discussion_r54285681>
> :
>
> >              # Save results to a csv file:
> >              output1.writerows(results1)
> >              output2.writerows(results2)
> >              output3.writerows(results3)
> > -
> > +
> > +    results = DataFrame(results, columns=['site', 'S', 'N', 'AICc_logseries', 'AICc_pln', 'AICc_negbin',
> > +                                         'AICc_zipf', 'likelihood_logseries', 'likelihood_pln',
> > +                                         'likelihood_negbin', 'likelihood_zipf', 'relative_ll_logseries',
> > +                                         'relative_ll_pln', 'relative_ll_negbin', 'relative_ll_zipf'])
> > +    results.to_csv(os.path.join(data_dir, dataset_name +  '_likelihood_results.csv'), index=False)
>
> Of course it would have been helpful to explain this rather complicated
> logic up front [image: :smile:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/weecology/sad-comparison/pull/189/files#r54285681>.
>